### PR TITLE
Add lint check that detects UI thread continuations.

### PR DESCRIPTION
**Pull request by vkryachko**

### CODE REVIEW

File: tools/lint/src/main/kotlin/TasksMainThreadDetector.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    /** Calling methods on the wrong thread */
+    @JvmField
+    val TASK_MAIN_THREAD =
+      Issue.create(
+        id = "TaskMainThread",
+        briefDescription = "Use an explicit Executor for Task continuations",
+        explanation =
+          """
+                    Not providing an executor results in continuations to execute on the main thread \
+                    which is not a good default. Please pass in an executor explicitly.
```
Comment (rlazo) [inline tools/lint/src/main/kotlin/TasksMainThreadDetector.kt:65]:
```suggestion
                    Not providing an executor results in continuations being executed on the main thread, \
                    which in most cases is not intended. Please pass in an executor explicitly.
```

File: tools/lint/src/main/kotlin/TasksMainThreadDetector.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package com.google.firebase.lint.checks
```
Comment (rlazo) [inline tools/lint/src/main/kotlin/TasksMainThreadDetector.kt:1]: qq, do we need the copyright header? I guess yes

File: tools/lint/src/main/kotlin/TasksMainThreadDetector.kt (hunk 1 above)
Comment (rlazo) [reply tools/lint/src/main/kotlin/TasksMainThreadDetector.kt:65]: Deep down we know is not a good default :-D, but let's try spin it differently.

**Review by rlazo — APPROVED** [review]

**Comment by vkryachko** [comment]
Oops, I prematurely merged, will send a follow up PR with suggested changed

**Pull request merged**